packages/graz/src/hooks/chains.ts

```diff
@@ -28,3 +28,23 @@ export function useSuggestChain({ onError, onLoading, onSuccess }: UseSuggestCha
     status: mutation.status,
   };
 }
+
+export type UseSuggestChainAndConnectArgs = MutationEventArgs<ChainInfo>;
+
+export function useSuggestChainAndConnect({ onError, onLoading, onSuccess }: UseSuggestChainAndConnectArgs = {}) {
+  const queryKey = ["WADESTA_USE_SUGGEST_CHAIN_AND_CONNECT", onError, onLoading, onSuccess];
+  const mutation = useMutation(queryKey, suggestChainAndConnect, {
+    onError: (err, chainInfo) => Promise.resolve(onError?.(err, chainInfo)),
+    onMutate: onLoading,
+    onSuccess: (chainInfo) => Promise.resolve(onSuccess?.(chainInfo)),
+  });
+
+  return {
+    error: mutation.error,
+    isLoading: mutation.isLoading,
+    isSuccess: mutation.isSuccess,
+    suggestAndConnect: mutation.mutate,
+    suggestAndConnectAsync: mutation.mutateAsync,
+    status: mutation.status,
```

Comment:
Sort props and add `isSupported` like on `useConnect`[^1]

[^1]: https://github.com/strangelove-ventures/graz/blob/1eca580fe58084369d6bfcb686c5f3d951118e6d/packages/graz/src/hooks/account.ts#L79-L85